### ISSUE TITLE
BigDL throws exceptions if it run on a platform whose dependency conflict with BigDL's

### ISSUE THREAD

**Issue by dding3** [intel-analytics/BigDL#1833]
Tried to run BigDL on customer's platform and found their dependent google protobuf is conflict with BigDL's. Although we have shaded protobuf, but our assembled jar still includes google protobuf which will cause the exception. Currently I need exclude them manually from bigdl jar to avoid the exception. I think it would be better if we exclude google protobuf when assembing bigdl jar.

**Comment by zhichao-li** [comment]
What if our model serialization require specified protobuf version? Shall we exclude protobuf explicitly from the client side if they dont need model saving.  cc @wzhongyuan

**Comment by wzhongyuan** [comment]
@dding3 @zhichao-li  There are 3 functionalities have dependency on protobuf : Caffe loader , Tensorflow loader and model serialization and protobuf was upgraded to 3.4 for tensorflow recently. So as long as users don't want to use any of these three, we can exclude them and only need it at compile time.

**Comment by jason-dai** [comment]
Shouldn't we use maven shade?

**Comment by dding3** [comment]
Yes, we used maven shade. However we didn't exclude google protobuf classes which are not needed from bigdl.jar. Now in bigdl.jar, we can find both shade protobuf and google protobuf. And google protobuf will cause the version conflict. We need manually excluded them to make sure bigdl can work. I think it would be better if we can change the way we assembed bigdl jar, exclude google protobuf while keep shade protobuf in to avoid google protobuf version conflicts.
